imqsauth/http.go

```diff
@@ -214,13 +223,18 @@ func (x *ImqsCentral) ResetPasswordStart(identity string, isNewAccount bool) (in
 		expireSeconds = x.Config.NewAccountExpirySeconds
 	}
 
-	token, err := x.Central.ResetPasswordStart(identity, time.Now().Add(time.Duration(expireSeconds)*time.Second))
+	token, err := x.Central.ResetPasswordStart(userId, time.Now().Add(time.Duration(expireSeconds)*time.Second))
 	if err != nil {
 		return http.StatusForbidden, "Error resetting password: " + err.Error()
 	}
 
 	// &welcome=true -- use this for welcome email
-	resetUrl, err := x.makeAbsoluteUrl("/#resetpassword=true&identity=" + url.QueryEscape(identity) + "&token=" + url.QueryEscape(token))
+	strUserId := strconv.FormatInt(int64(userId), 10)
+	identity, errGetIdentity := x.Central.GetIdentityFromUserId(userId)
+	if errGetIdentity != nil {
+		return http.StatusForbidden, "Error sending mail: " + err.Error()
```

Comment:
I think this should have a different error response, such as `return http.StatusBadRequest, "Unrecognized userid: "...`